### PR TITLE
feat: support external service

### LINKED ISSUE #927: Proposal: support external name service
I have seen some voices in the community, hoping that APISIX Ingress can proxy external services e.g: #813 #645 

For these two types of requirements, it is a relatively simple requirement for #645, we only need to add the corresponding External name type service to complete.

But for #645 , I found a very interesting situation. No other Ingress controller implements similar functionality yet, and I think this would be a huge feature.

APISIX actually supports setting the domain name to nodes in the upstream. But APISIX Ingress is not yet supported.

To achieve the above function, we can set a special `resolveGranularity` to directly convert the record of the external name to Node.

### PR DESCRIPTION
**Pull request by lingsamuel**
Signed-off-by: Ling Samuel <lingsamuelgrace@gmail.com>

### Type of change:

- [x] New feature provided

### What this PR does / why we need it:
First part of #927 
Proposal is here: https://github.com/apache/apisix-ingress-controller/issues/927#issuecomment-1206116262

This feature is considered **unstable** because how to implement some features (schemes and ports) is still under discussion.

Doc PR is incoming.

### CODE REVIEW

**Comment by tao12345666333** [comment]
This is a new implementation of #726 and #927  , PTAL.

/cc @elvis-cai  @purekeeper @BrandonArp @NMichas

**Comment by lingsamuel** [comment]
All tests passed, just a linter error.

**Comment by tao12345666333** [comment]
could you please resolve conflicts? thanks

**Review by AlinsRan — APPROVED** [review]

File: pkg/providers/apisix/translation/apisix_upstream.go [hunk 1]
```diff
@@ -47,3 +56,116 @@ func (t *translator) translateService(namespace, svcName, subset, svcResolveGran
...
+					n.Port = 80
+				}
+			} else if len(arr) == 2 {
+				port, err := strconv.Atoi(arr[1])
+				if err != nil {
+					return nil, errors.Wrap(err, fmt.Sprintf("failed to parse ApisixUpstream %s/%s port: at ExternalNodes[%v]: %s", au.Namespace, au.Name, i, node.Name))
+				}
+
+				n.Port = port
+			}
```
Comment (tao12345666333) [inline pkg/providers/apisix/translation/apisix_upstream.go:130]: ExternalName Service is a DNS CNAME record.
According to RFC1123, it does not include protocols such as HTTP/HTTPS and ports

Comment (lingsamuel) [reply pkg/providers/apisix/translation/apisix_upstream.go:130]: APISIX requires a port and the port field has not been implemented yet. This is a temporary solution.

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**